sound/soc/sof/intel/hda.c

```diff
@@ -290,8 +297,26 @@ static int hda_init(struct snd_sof_dev *sdev)
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
 
+static int check_nhlt_dmic(struct snd_sof_dev *sdev)
+{
+	struct nhlt_acpi_table *nhlt;
+	int dmic_num;
+
+	nhlt = intel_nhlt_init(sdev->dev);
+	if (nhlt) {
+		dmic_num = intel_nhlt_get_dmic_geo(sdev->dev, nhlt);
+		intel_nhlt_free(nhlt);
+		if (dmic_num == 2 || dmic_num == 4)
```

Comment:
This seems a bit odd. If this is a true hw limit, then ok.

Reply:
@singalsu can you chime in? I don't think we have any topologies that support 1 or 3 channels.
also the only definitions I see in NHLT are 2 and 4 channels.

Reply:
Sorry for delay, I missed this. We do support any 1 mic with topology (there's MONO_PDM0_MICA, MONO_PDM0_MICB, and also MONO_PDM1_MICx is possible but macros are missing).

Generally also 3 microphones could be possible but not with current Intel platforms. If we have such device need to capture 4 channels and use channel map to extract valid channels.

Reply:
Is this something we can test with the Up2 DMIC kit? For now I have no means of testing 1mic support.
I actually wonder if anyone would do this?